python/foglamp/services/south/server.py

```diff
@@ -230,9 +233,14 @@ def run(self):
             # Cancel all pending asyncio tasks after a timeout occurs
             done, pending = await asyncio.wait(asyncio.Task.all_tasks(), timeout=_CLEAR_PENDING_TASKS_TIMEOUT)
             for task_pending in pending:
-                task_pending.cancel()
+                try:
+                     task_pending.cancel()
+                     # Now, await task  to execute it's cancellation.
+                     # Cancelled task raises asyncio.CancelledError:
+                except asyncio.CancelledError:
+                    pass
```

Comment:
cancelled error can be genuine, should we await task? or we don;t care as we are going to stop it anyway, as we have already provided _CLEAR_PENDING_TASKS_TIMEOUT.

Reply:
If you want to handle CancelledError to do some local processing or logging or whatever, you need to catch this exception within future/task itself and add the handler code in the except block locally. south/server.py->stop() is not the right place for it. Secondly, cancellation of all user defined tasks/futures have already been handled prior to this code. This code block is just a kind of safety check to ensure that cancellation of remaining pending tasks, if any at all, are properly taken care of.